services/related.py

```
@@ -0,0 +1,364 @@
+                                        "path": f"/blog/{digest['date']}"
+                                    }
+                                    posts.append(post_info)
+                                    logger.info(f"Found remote post: {date_str} - {post_info['title']}")
+                                else:
+                                    logger.debug(f"Skipping digest {date_str} - not v2 or missing frontmatter")
+                            elif digest_response.status_code == 404:
+                                logger.debug(f"No digest found for {date_str}")
+                            else:
+                                logger.debug(f"Failed to fetch digest for {date_str}: HTTP {digest_response.status_code}")
+                                    
+                        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)